UnityProject/Assets/Scripts/Electricity/PowerSupplies/PowerGenerator.cs

```diff
@@ -164,6 +163,11 @@ public void ServerPerformInteraction(HandApply interaction)
 	{
 		if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.SolidPlasma))
 		{
+			if (itemSlot.Item)
+			{
+				Chat.AddWarningMsgFromServer(interaction.Performer, "The generator already contains a sheet!");
+				return;
+			}
```

Comment:
The generator uses ItemStorage now, so it should be able to contain up to fifty sheets.

Reply:
it doesn't stack for some reason

Reply:
ye im making it stack ETA a couple hours